contracts/cw721-base/src/contract.rs

```diff
@@ -478,12 +484,14 @@ fn query_all_approvals<S: Storage, A: Api, Q: Querier>(
     start_after: Option<HumanAddr>,
     limit: Option<u32>,
 ) -> StdResult<ApprovedForAllResponse> {
-    let owner_raw = deps.api.canonical_address(&owner)?;
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = calc_range_start_human(deps.api, start_after)?;
+    let start_canon = maybe_canonical(deps.api, start_after)?;
+    let start = OwnedBound::exclusive(start_canon);
 
-    let res: StdResult<Vec<_>> = operators_read(&deps.storage, &owner_raw)
-        .range(start.as_deref(), None, Order::Ascending)
+    let owner_raw = deps.api.canonical_address(&owner)?;
+    let res: StdResult<Vec<_>> = OPERATORS
+        .prefix(&owner_raw)
+        .range(&deps.storage, start.bound(), Bound::None, Order::Ascending)
```

Comment:
`OwnedBound` may  be easier to use in some cases, but having two structs for one purpose requires more helpers, like the `bound()` method above. I fr one would try to stick to one `Bound`, unless really needed / justified.

Reply:
Fair enough. I can rename `OwnedBound` to bound. Only issue is when writing tests it is more verbose... 

`Bound::Exclusive(b"foo")` -> `Bound::Exclusive(b"foo".to_vec())`

which is something I hit quite a bit, and tried to "clean up". Then again, the actual usage is easier the way you describe and maybe best I optimize for production code, not my tests :smile:

Reply:
Yep. Though clarity of the API should reflect in unit tests, though. Maybe there's some syntactic sugar equivalent to `to_vec()` around?